runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -1486,6 +1557,12 @@ void amHandler(void* argNil) {
       DBG_PRINTF(DBG_ACK, "tx ack counter %d", count);
       tcip->numTxsOut -= count;
     }
+
+    {
+      static __thread int progressInterval;
+      if ((++progressInterval & 0x100) != 0)
```

Comment:
What's the rational for this? As-is I think it's calling `chpl_comm_make_progress` every iteration after the first 255 attempts, which seems a little odd or specific.

Reply:
It's providing backstop support for the non-AM handlers which are primarily responsible for progress (though the comm layer itself is only responsible for progress when we are doing manual).  So it's an "every now and then" thing.  We need this for the case where what we're progressing is the PUT of an AM 'done' indicator from a remote node back to us, when none of our non-AM handler threads are doing communication and thus supporting progress.  The AM handler is the only candidate in this case.  Should I add a comment?

Reply:
The part I'm not following is why it's every time after 256 iters, instead of once in every 256 iters. i.e. why `((++progressInterval & 0x100) != 0)` instead of `((++progressInterval & 0xFF) == 0)`

Reply:
Because I'm an idiot, of course.  Although it's not every time after 256.  It looks like it's "don't do it for 256 iters, then do do it for the next 256, then not for the 256 following that, then ...".